edenai_apis/apis/deepl/deepl_api.py

```diff
@@ -48,13 +48,13 @@ def translation__automatic_translation(
         }
 
         response = requests.request("POST", url, headers=self.header, data=data)
-        original_response = response.json()
-
-        if response.status_code >= 500:
+        try:
+            original_response = response.json()
+        except json.JSONDecodeError as exc:
             raise ProviderException(
-                message=http.client.responses[response.status_code],
-                code=response.status_code,
-            )
+                message=response.text, code=response.status_code
+            ) from exc
+
         if response.status_code != 200:
             raise ProviderException(
                 message=original_response["message"], code=response.status_code
```

Comment:
should not we keep the check for 500 also ? because if it's 500 error code, I don't think the original response will have a `message` key, so the `original_response["message"]` will not be accecible

Reply:
usually for deepl if its a 500 you have a json decode error

Reply:
I mean for cases where it's a 500 error code, but not a jsonDecodeError !